gammapy/catalog/fermi.py

```diff
@@ -1648,6 +1658,91 @@ def _info_spectral_fit(self):
+            Map containing the best fit.
+        """
+        table = Table.read(self._auxiliary_filename, hdu="BEST_FIT_LC")
+
+        # For best-fit profile, Ph_min and Ph_max are equal and represent bin centers.
+        phases = MapAxis.from_nodes(table["Ph_Min"], name="phase", interp="lin")
+        best_fit_profile = RegionNDMap.create(
+            region=None, axes=[phases], data=table["Intensity"]
+        )
+        return best_fit_profile
+
+    @property
+    def pulse_profile_radio(self):
+        """
+        Radio pulse profile provided in the auxiliary file of 3PC.
+
+        Returns
+        -------
+
+        radio_profile: `~gammapy.maps.RegionNDMap`
+            Map containing the radio profile.
+        """
+        table = Table.read(self._auxiliary_filename, hdu="RADIO_PROFILE")
+
+        # Need to do this because some PSR (J0540-6919) has duplicates
+        ph_node, unique_idx = np.unique(table["Ph_Min"], return_index=True)
+        data = table["Norm_Intensity"][unique_idx]
+
+        # For radio pulse profile, Ph_min and Ph_max are equal and represent bin centers.
+        phases = MapAxis.from_nodes(ph_node, name="phase", interp="lin")
+        radio_profile = RegionNDMap.create(region=None, axes=[phases], data=data)
+        return radio_profile
+
+    @property
+    def pulse_profiles(self):
+        """
+        The 3PC pulse profiles are provided in different energy ranges, each represented in weighted counts.
+        These profiles are stored in a dictionary of `~gammapy.maps.RegionNDMap` objects, one per energy bin.
+
+        The dictionary keys correspond to specific energy ranges as follows:
+
+        - `GT100_WtCnt`: > 0.1 GeV
+        - `50_100_WtCt`: 0.05 – 0.1 GeV
+        - `100_300_WtCt`: 0.1 – 0.3 GeV
+        - `300_1000_WtCt`: 0.3 – 1 GeV
+        - `1000_3000_WtCt`: 1 – 3 GeV
+        - `3000_100000_WtCt`: 3 – 1000 GeV
+        - `10000_100000_WtCt`: 10 – 1000 GeV
+
+        Each pulse profile has an associated uncertainty map, which can be accessed by
+        prepending `"Unc_"` to the corresponding key in the dictionary.
+
+        Returns
+        -------
+
+        map_dict: dict of `~gammapy.maps.RegionNDMap`
+            Dictionary of map containing the pulse profile in different energy bin.
+        """
+
+        table = Table.read(self._auxiliary_filename, hdu="GAMMA_LC")
+        phases = MapAxis.from_edges(
+            np.unique(np.concatenate([table["Ph_Min"], table["Ph_Max"]])),
+            name="phase",
+            interp="lin",
+        )
+        map_dict = dict()
+        for name in self._pulse_profile_column_name:
+            map_dict[name] = RegionNDMap.create(
+                region=None, axes=[phases], data=table[name]
+            )
+        for name in self._pulse_profile_column_name:
+            map_dict[f"Unc_{name}"] = RegionNDMap.create(
+                region=None, axes=[phases], data=table[f"Unc_{name}"]
+            )
+        return map_dict
```

Comment:
Here you could use the `gammapy.maps.Maps` as all your maps have the same geometry.

You could also have used `LabelMapAxis` to represent your no continuous energy bands. It might more difficult to interact with for little gain though.

Reply:
I forgot about `Maps`, that's a good very good idea. 
Should I also use it for the radio and best fit profile for consistency ?